core/lib/analytics/react/index.tsx

```diff
@@ -0,0 +1,24 @@
+'use client';
+
+import { createContext, useContext, useEffect } from 'react';
+
+import { type Analytics } from '../types';
+
+const AnalyticsContext = createContext<Analytics | null>(null);
+
+interface AnalyticsProviderProps {
+  analytics: Analytics | null;
+  children: React.ReactNode;
+}
+
+export const AnalyticsProvider = ({ children, analytics }: AnalyticsProviderProps) => {
+  useEffect(() => {
+    analytics?.initialize();
```

Comment:
Should we have `initialize` return a cleanup function or is that something that most analytics providers don't provide? I'm guessing that it being a singleton covers any reinitialization issue that could occur?

Reply:
I didn't find a need so left it out. If we do see a situation where a cleanup is need I would add it, but yeah the singleton + browsers having to remount analytics solves most of our use-cases.